packages/ymax-planner/src/engine.ts

```diff
@@ -0,0 +1,401 @@
+  try {
+    console.warn('\n[Engine] Testing Spectrum pool balance API...');
+    const testEthAddress = '0x87870Bca3F3fD6335C3F4ce8392D69350B4fA4E2'; // Example ETH address
+
+    const testCases = [{ chain: 'ethereum', pool: 'aave' }];
+
+    for (const testCase of testCases) {
+      try {
+        console.warn(
+          `\n[Engine] Testing ${testCase.chain}/${testCase.pool} pool balance for ${testEthAddress}`,
+        );
+        const poolBalance = await spectrum.getPoolBalance(
+          testCase.chain as any,
+          testCase.pool as any,
+          testEthAddress,
+        );
+        console.warn(`[Engine] ✅ Pool balance found:`, {
+          supplyBalance: poolBalance.balance.supplyBalance.toLocaleString(),
+          borrowAmount: poolBalance.balance.borrowAmount.toLocaleString(),
+          chain: poolBalance.chain,
+          pool: poolBalance.pool,
+        });
+      } catch (specError) {
+        console.warn(
+          `[Engine] Pool balance test (${testCase.chain}/${testCase.pool}): ${specError instanceof Error ? specError.message : specError}`,
+        );
+      }
+    }
+  } catch (error) {
+    console.error('[Engine] Spectrum balance testing failed:', error);
+  }
+
+  // Test Cosmos REST API client with Noble chain
+  try {
+    console.warn('[Engine] Testing Cosmos REST API client...');
+    const availableChains = cosmosRest.getAvailableChains();
+    console.warn(
+      `[Engine] Available chains: ${availableChains.map(c => c.config.name).join(', ')}`,
+    );
+
+    // Test chain info for Noble
+    const nobleInfo = await cosmosRest.getChainInfo('noble');
+    console.warn(
+      `[Engine] Noble chain info retrieved successfully. Chain ID: ${(nobleInfo as any)?.default_node_info?.network || 'unknown'}`,
+    );
+
+    // Test balance fetching for a known address (this will fail gracefully if address doesn't exist)
+    // Using a placeholder address - in real usage, this would be a user's address
+    const testAddress = 'noble1xw2j23rcwrkg02yxdn5ha2d2x868cuk6370s9y'; // This is just for testing the API structure
+    try {
+      const balances = await cosmosRest.getAccountBalances(
+        'noble',
+        testAddress,
+      );
+      console.warn(
+        `[Engine] Noble balance API test successful. Response structure verified. Found ${balances.balances.length} balance(s):`,
+        balances.balances
+          .map(coin => `${coin.amount} ${coin.denom}`)
+          .join(', ') || 'No balances',
+      );
+    } catch (balanceError) {
+      // This is expected to fail with the test address, but shows the API is working
+      console.warn(
+        `[Engine] Noble balance API responding (expected error for test address): ${balanceError instanceof Error ? balanceError.message : balanceError}`,
+      );
+    }
+  } catch (error) {
+    console.error('[Engine] Cosmos REST API initialization failed:', error);
+    // Continue with limited functionality
+  }
```

Comment:
```suggestion
  // Test Cosmos REST API client with Noble chain
  try {
    const nobleInfo = await cosmosRest.getChainInfo('noble');
    console.warn(
      'Noble chain ID',
      (nobleInfo as any)?.default_node_info?.network,
      nobleInfo,
    );

    // Test balance fetching for a known address (this will fail gracefully if
    // address doesn't exist)
    const testAddress = 'noble1xw2j23rcwrkg02yxdn5ha2d2x868cuk6370s9y';
    const balances = await cosmosRest.getAccountBalances('noble', testAddress);
    console.warn(
      'Noble balance denoms',
      balances.balances.map(coin => coin.denom),
    );
  } catch (err) {
    console.error('Noble getAccountBalances failed', err);
    throw err;
  }
```